dashevo/mn-bootstrap
# feat: add platform:dapi:ssl:obtain command

Adds the command platform:dapi:ssl:obtain command

## Issue being fixed or feature implemented
This command automate the process to obtain an IP cert from ZeroSSL using the ZeroSSL REST API

## What was done?
Implemented the REST API in `src/ssl/zerossl`
The command platform:dapi:ssl:obtain does the following steps:

1. Create a cert and save the challenge file
2. Setup a temp server the verify the IP ownership
3. Test if the temp server is up
4. Verify the IP ownership
5. Download and save the cert file

## How Has This Been Tested?
1. Install the temp server command 
```
npm install --global http-server
```
2. Generate the CSR and private key
```
cd configs/evonet/dapi/nginx/
openssl req -new -newkey rsa:2048 -nodes -keyout private.key -out domain.csr
```
3. Run the command mn platform:dapi:ssl:obtain
```
cd ~/mn-bootstrap
USAGE
  $ mn platform:dapi:ssl:obtain PRESET EXTERNAL-IP ZEROSSL-APIKEY

ARGUMENTS
  PRESET          (local|testnet|evonet) preset to use
  EXTERNAL-IP     masternode external IP
  ZEROSSL-APIKEY  ZeroSSL API Key - https://app.zerossl.com/developer
```
